nvtabular/ops/normalize.py

```diff
@@ -46,24 +47,31 @@ def __init__(self):
         super().__init__()
         self.means = {}
         self.stds = {}
+        self.means_cupy = self.stds_cupy = None
 
     @annotate("Normalize_fit", color="green", domain="nvt_python")
     def fit(self, columns: ColumnNames, ddf: dask_cudf.DataFrame):
         return _custom_moments(ddf[columns])
 
     def fit_finalize(self, dask_stats):
+        self.means_cupy = cupy.array(dask_stats["mean"].values)
+        self.stds_cupy = cupy.array(dask_stats["std"].values)
+
         for col in dask_stats.index:
             self.means[col] = float(dask_stats["mean"].loc[col])
             self.stds[col] = float(dask_stats["std"].loc[col])
 
     @annotate("Normalize_op", color="darkgreen", domain="nvt_python")
     def transform(self, columns: ColumnNames, gdf: cudf.DataFrame) -> cudf.DataFrame:
-        new_gdf = cudf.DataFrame()
-        for name in columns:
-            if self.stds[name] > 0:
-                new_gdf[name] = (gdf[name] - self.means[name]) / (self.stds[name])
-                new_gdf[name] = new_gdf[name].astype("float32")
-        return new_gdf
+        arr = cupy.fromDlpack(gdf[columns].astype("float32").to_dlpack())
+        arr -= self.means_cupy
+        arr /= self.stds_cupy
+        ret = cudf.from_dlpack(arr.toDlpack())
+        if isinstance(ret, cudf.Series):
+            ret = cudf.DataFrame({columns[0]: ret})
+        else:
+            ret.columns = columns
```

Comment:
(with cpu/gpu dispatching in mind) Maybe something like this will work:  `type(gdf)(arr, columns=columns)`